File: langs/python/testdata/fixture.py
from enum import Enum
from typing import Protocol
"""Fixture exercising inheritance, decorators, nesting and async."""

MAX_ANIMALS = 100
default_name = "rex"


class Animal:
    """Base animal.

    Second paragraph, not extracted.
    """

    kingdom = "animalia"
    count: int = 0

    def __init__(self, name):
        self.name = name
        self.legs: int = 4

    def speak(self):
        """Make a sound."""
        raise NotImplementedError


class Dog(Animal):
    def speak(self):
        return "woof"

    @property
    def tricks(self):
        return ["fetch"]

    @staticmethod
    def species():
        return "canis"

    class Puppy:
        """A nested class."""

        def nap(self):
            pass


class Puppy(Dog, Animal):
    pass


async def feed(animal):
    """Feed an animal, asynchronously."""
    await animal.eat()


def make_sound(a):
    return a.speak()


def outer():
    def inner():
        return 1

    return inner()


class Drawable(Protocol):
    def draw(self): ...


class Color(Enum):
    RED = 1
    GREEN = 2

File: langs/golang/testdata/fixture.go
package fixture

// MaxAnimals is the herd cap.
const MaxAnimals = 100

var defaultName = "rex"

// Speaker is anything that speaks.
type Speaker interface {
	Speak() string
}

// Animal is a base struct.
type Animal struct {
	Name string
}

// Dog embeds Animal.
type Dog struct {
	Animal
	Tricks []string
}

// Speak returns the animal's sound.
func (a Animal) Speak() string { return "..." }

// NewAnimal builds an Animal.
func NewAnimal(name string) *Animal { return &Animal{Name: name} }
